tests/e2e/e2e_simple_test.go

```diff
@@ -1,114 +1,157 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"time"
 )
 
 var _ = Describe("ISO build test", func() {
-	//k := kubectl.New()
-	Context("registration", func() {
+	k8s := dynamic.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	scheme := runtime.NewScheme()
+	_ = osbuilder.AddToScheme(scheme)
+
+	var artifactName string
+	artifacts := k8s.Resource(schema.GroupVersionResource{Group: osbuilder.GroupVersion.Group, Version: osbuilder.GroupVersion.Version, Resource: "osartifacts"}).Namespace("default")
+	pods := k8s.Resource(schema.GroupVersionResource{Group: corev1.GroupName, Version: corev1.SchemeGroupVersion.Version, Resource: "pods"}).Namespace("default")
+	pvcs := k8s.Resource(schema.GroupVersionResource{Group: corev1.GroupName, Version: corev1.SchemeGroupVersion.Version, Resource: "persistentvolumeclaims"}).Namespace("default")
+	jobs := k8s.Resource(schema.GroupVersionResource{Group: batchv1.GroupName, Version: batchv1.SchemeGroupVersion.Version, Resource: "jobs"}).Namespace("default")
+
+	artifact := &osbuilder.OSArtifact{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "OSArtifact",
+			APIVersion: osbuilder.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "simple-",
+		},
+		Spec: osbuilder.OSArtifactSpec{
+			ImageName: "quay.io/kairos/core-opensuse:latest",
+			ISO:       true,
+			DiskSize:  "",
+			Exporters: []batchv1.JobSpec{
+				{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							RestartPolicy: corev1.RestartPolicyNever,
+							Containers: []corev1.Container{
+								{
+									Name:    "test",
+									Image:   "debian:latest",
+									Command: []string{"bash"},
+									Args:    []string{"-xec", "[ -f /artifacts/*.iso ]"},
+									VolumeMounts: []corev1.VolumeMount{
+										{
+											Name:      "artifacts",
+											ReadOnly:  true,
+											MountPath: "/artifacts",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	uArtifact := unstructured.Unstructured{}
+	uArtifact.Object, _ = runtime.DefaultUnstructuredConverter.ToUnstructured(artifact)
+	resp, err := artifacts.Create(context.TODO(), &uArtifact, metav1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	artifactName = resp.GetName()
+
+	Context("simple", func() {
+		artifactLabelSelectorReq, _ := labels.NewRequirement("build.kairos.io/artifact", selection.Equals, []string{artifactName})
+		artifactLabelSelector := labels.NewSelector().Add(*artifactLabelSelectorReq)
+
+		It("starts the build", func() {
+			Eventually(func(g Gomega) {
+				w, err := pods.Watch(context.TODO(), metav1.ListOptions{LabelSelector: artifactLabelSelector.String()})
+				Expect(err).ToNot(HaveOccurred())
+
+				var stopped bool
+				for !stopped {
+					event, ok := <-w.ResultChan()
 
-		AfterEach(func() {
-			kubectl.New().Delete("osartifacts", "-n", "default", "hello-kairos")
+					stopped = event.Type != watch.Deleted && event.Type != watch.Error || !ok
+				}
+			}).Should(Succeed())
```

Comment:
maybe needs a timeout? just in case

Reply:
and in the other cases below

Reply:
I agree, I was relying on the gh actions timeout for ci but locally it spins forever.